src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -79,17 +85,25 @@ public AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryI
             : PartitionAdditionalFiles(allFiles, analysisConfig);
     }
 
-    private List<FileInfo> GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
+    private FileInfo[] GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
         directoryWrapper
-            .EnumerateFiles(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
+            .EnumerateDirectories(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
+            .Concat([projectBaseDir.FullName]) // also include the root directory
+            .Where(x => !IsExcludedDirectory(x))
+            .SelectMany(x => directoryWrapper.EnumerateFiles(x, "*", SearchOption.TopDirectoryOnly))
             .Select(x => new FileInfo(x))
             .Where(x => extensions.Any(e => x.Name.EndsWith(e, StringComparison.OrdinalIgnoreCase) && !x.Name.Equals(e, StringComparison.OrdinalIgnoreCase)))
-            .ToList();
+            .ToArray();
```

Comment:
Can you use `ToImmutableArray` here? If so, it is preferred over an array.

Reply:
No, I cannot, no reference.